servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/DefaultExecutor.java

```diff
@@ -249,11 +254,27 @@ public Cancellable schedule(final Runnable task, final long delay, final TimeUni
             // When using the global scheduler, offload timer ticks to the user specified Executor since user code
             // executed on the timer tick can block.
             ScheduledFuture<?> future = GLOBAL_SINGLE_THREADED_SCHEDULED_EXECUTOR.schedule(
-                    () -> offloadExecutor.execute(task), delay, unit);
-            // Schedulers are only used to generate a tick and do not execute any user code. This means they will never
-            // run any blocking code and hence it does not matter whether we use the interruptOnCancel as sent by the
-            // user upon creation in the scheduler. User code (completion of Completable on tick) will be executed on
-            // the configured executor and not the Scheduler thread.
+                    () -> {
+                        try {
+                            offloadExecutor.execute(task);
+                        } catch (Throwable t) {
+                            LOGGER.error("Unexpected exception while offloading scheduled task: {} to executor: {}.",
+                                    task, offloadExecutor, t);
```

Comment:
```suggestion
                            LOGGER.error("Unexpected exception while offloading scheduled task: {} to executor: {}. " +
                                            "Will execute the task on {} thread.",
                                    task, offloadExecutor, Thread.currentThread().getName(), t);
```

WDYT? You can commit from the UI, I verified locally.